In [3]:
import pandas as pd  # for working with data tables
from sklearn.model_selection import train_test_split  #Tells from where to import the train_test_split
from sklearn.tree import DecisionTreeClassifier, plot_tree  # for building decision trees
from sklearn.metrics import accuracy_score, classification_report 
import matplotlib.pyplot as plt  # Imports matplotlib

print("✓ Libraries imported successfully!")

✓ Libraries imported successfully!


In [4]:
import kagglehub
path = kagglehub.dataset_download("sakshisatre/titanic-dataset")
print("Dataset downloaded to:", path)

Dataset downloaded to: C:\Users\mc_glucius26\.cache\kagglehub\datasets\sakshisatre\titanic-dataset\versions\2


In [5]:
df = pd.read_csv("Titanic Dataset.csv")  # Make sure your csv 'Titanic Dataset.csv' file is in the right folder!

print("✓ Dataset loaded successfully!")
print(f"Number of rows: {len(df)}")
print(f"Number of columns: {len(df.columns)}")

✓ Dataset loaded successfully!
Number of rows: 1309
Number of columns: 14


In [6]:
df.head()

pclass  survived                                             name     sex  \
0       1         1                    Allen, Miss. Elisabeth Walton  female   
1       1         1                   Allison, Master. Hudson Trevor    male   
2       1         0                     Allison, Miss. Helen Loraine  female   
3       1         0             Allison, Mr. Hudson Joshua Creighton    male   
4       1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

     age  sibsp  parch  ticket      fare    cabin embarked boat   body  \
0  29.00      0      0   24160  211.3375       B5        S    2    NaN   
1   0.92      1      2  113781  151.5500  C22 C26        S   11    NaN   
2   2.00      1      2  113781  151.5500  C22 C26        S  NaN    NaN   
3  30.00      1      2  113781  151.5500  C22 C26        S  NaN  135.0   
4  25.00      1      2  113781  151.5500  C22 C26        S  NaN    NaN   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

In [8]:
df.tail()      

pclass  survived                       name     sex   age  sibsp  parch  \
1304       3         0       Zabour, Miss. Hileni  female  14.5      1      0   
1305       3         0      Zabour, Miss. Thamine  female   NaN      1      0   
1306       3         0  Zakarian, Mr. Mapriededer    male  26.5      0      0   
1307       3         0        Zakarian, Mr. Ortin    male  27.0      0      0   
1308       3         0         Zimmerman, Mr. Leo    male  29.0      0      0   

      ticket     fare cabin embarked boat   body home.dest  
1304    2665  14.4542   NaN        C  NaN  328.0       NaN  
1305    2665  14.4542   NaN        C  NaN    NaN       NaN  
1306    2656   7.2250   NaN        C  NaN  304.0       NaN  
1307    2670   7.2250   NaN        C  NaN    NaN       NaN  
1308  315082   7.8750   NaN        S  NaN    NaN       NaN

In [9]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 14 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   pclass     1309 non-null   int64  
 1   survived   1309 non-null   int64  
 2   name       1309 non-null   object 
 3   sex        1309 non-null   object 
 4   age        1046 non-null   float64
 5   sibsp      1309 non-null   int64  
 6   parch      1309 non-null   int64  
 7   ticket     1309 non-null   object 
 8   fare       1308 non-null   float64
 9   cabin      295 non-null    object 
 10  embarked   1307 non-null   object 
 11  boat       486 non-null    object 
 12  body       121 non-null    float64
 13  home.dest  745 non-null    object 
dtypes: float64(3), int64(4), object(7)
memory usage: 143.3+ KB
None


In [10]:
# TODO: Write code to find the total number of rows
total_passengers = 1309
print(f"Total passengers in dataset: {total_passengers}")

Total passengers in dataset: 1309


In [11]:
print(df.columns)

Index(['pclass', 'survived', 'name', 'sex', 'age', 'sibsp', 'parch', 'ticket',
       'fare', 'cabin', 'embarked', 'boat', 'body', 'home.dest'],
      dtype='object')


In [15]:
# TODO: Write code to display all column names
print(f"Column names: {df.columns}")
print(df.columns)  # Hint: use .columns

Column names: Index(['pclass', 'survived', 'name', 'sex', 'age', 'sibsp', 'parch', 'ticket',
       'fare', 'cabin', 'embarked', 'boat', 'body', 'home.dest'],
      dtype='object')
Index(['pclass', 'survived', 'name', 'sex', 'age', 'sibsp', 'parch', 'ticket',
       'fare', 'cabin', 'embarked', 'boat', 'body', 'home.dest'],
      dtype='object')


In [16]:
print(df.describe())

            pclass     survived          age        sibsp        parch  \
count  1309.000000  1309.000000  1046.000000  1309.000000  1309.000000   
mean      2.294882     0.381971    29.881138     0.498854     0.385027   
std       0.837836     0.486055    14.413493     1.041658     0.865560   
min       1.000000     0.000000     0.170000     0.000000     0.000000   
25%       2.000000     0.000000    21.000000     0.000000     0.000000   
50%       3.000000     0.000000    28.000000     0.000000     0.000000   
75%       3.000000     1.000000    39.000000     1.000000     0.000000   
max       3.000000     1.000000    80.000000     8.000000     9.000000   

              fare        body  
count  1308.000000  121.000000  
mean     33.295479  160.809917  
std      51.758668   97.696922  
min       0.000000    1.000000  
25%       7.895800   72.000000  
50%      14.454200  155.000000  
75%      31.275000  256.000000  
max     512.329200  328.000000  




Q1: How many passengers are in the dataset?

A: 1309

Q2: What is the average (mean) age of passengers?

A:29.88113

Q3: What is the minimum fare paid? Maximum fare paid?

A:0.000000, 512.329200


In [17]:
# TODO: Write code to count survivors
survivors = df['survived'].sum()  # sum() adds up all the 1s
print(f"Number of survivors: {survivors}")

Number of survivors: 500


500

In [18]:
# TODO: Calculate survival rate
survival_rate = (survivors / total_passengers) * 100
print(f"Survival rate: {survival_rate:.2f}%")

Survival rate: 38.20%


38.20%

In [19]:
df.isnull().sum()

pclass          0
survived        0
name            0
sex             0
age           263
sibsp           0
parch           0
ticket          0
fare            1
cabin        1014
embarked        2
boat          823
body         1188
home.dest     564
dtype: int64

body